src/writeckpt.cpp

```diff
@@ -207,7 +207,7 @@ void mtcp_writememoryareas(int fd)
      */
 
     if (!((area.prot & PROT_READ) || (area.prot & PROT_WRITE)) &&
-        area.name[0] != '\0') {
+        (area.name[0] != '\0') && strcmp(area.name, "[heap]")) {
```

Comment:
Let's make this:
```C
(area.name[0] != '\0') && strcmp(area.name, "[heap]") && \
strcmp(area.name, "[stack]") && (Util::strStartsWith(area.name, "[stack:"
```
to make this a little more future-proof.  In addition to the heap, this could happen to the stack or to a secondary thread stack.